src/services/chain.service.ts

```diff
@@ -0,0 +1,248 @@
+                // We iterate over all the clients and store them in an array of LumClient[]
+                this._chainClient = await LumClient.connect(this._configService.get<string>(env));
+                const chainId = await this._chainClient.getChainId();
+
+                this._client.push(this._chainClient);
+
+                this._logger.log(`Connection established to ${env} ${this._configService.get<string>(env)} = ${chainId}`);
+            }
+        } catch (e) {
+            console.error(e);
+
+            Sentry.captureException(e);
+        }
+    };
+
+    isInitialized = (): boolean => {
+        return this._client !== null;
+    };
+
+    getPrice = async (): Promise<{ unit_price_usd: number; symbol: string }[]> => {
+        try {
+            // Observable to get the prices and filter with the one we represent in the index
+            const price = await lastValueFrom(this._httpService.get(`https://api-osmosis.imperator.co/tokens/v2/all`).pipe(map((response) => response.data)));
+
+            // We exclude LUM's price as we have our seperate service for that
+            return price
+                .filter((el) => this._assetSymbol.some((f) => f === el.symbol && el.symbol !== AssetSymbol.LUM))
+                .map((el) => ({
+                    unit_price_usd: el.price,
+                    symbol: el.symbol,
+                }));
+        } catch (error) {
+            this._logger.error(`Could not fetch Price Price...`, error);
+
+            Sentry.captureException(error);
+
+            return null;
+        }
+    };
+
+    getMcap = async (): Promise<{ total_value_usd: number; symbol: string }[]> => {
+        try {
+            // Observable to get the mcap and filter with the one we represent in the index
+            const getMktCap = await lastValueFrom(this._httpService.get(`https://api-osmosis.imperator.co/tokens/v2/mcap`).pipe(map((response) => response.data)));
+
+            // We exclude LUM's mcap as we have our seperate service for that
+            return getMktCap
+                .filter((el) => this._assetSymbol.some((f) => f === el.symbol && el.symbol !== AssetSymbol.LUM))
+                .map((el) => ({
+                    total_value_usd: el.market_cap,
+                    symbol: el.symbol,
+                }));
+        } catch (error) {
+            this._logger.error(`Could not fetch Market Cap for External Chains...`, error);
+
+            Sentry.captureException(error);
+
+            return null;
+        }
+    };
+
+    getTokenSupply = async (): Promise<{ supply: number; symbol: string }[]> => {
+        try {
+            // Not all chains have a reliable token supply that we can extract from chain
+            // Hence for some, we need to rely on their official endpoint to retrieve it
+
+            // Return token supply based on the microDenum and denum from assets in the index
+            const chainSupply = await Promise.all(
```

Comment:
redundant promise

Reply:
It's the only place with getChainApy where we need this redundancy. For the rest it has been corrected.